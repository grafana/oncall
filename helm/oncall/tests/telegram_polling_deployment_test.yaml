suite: test telegram polling deployment
templates:
  - telegram-polling/deployment.yaml
release:
  name: oncall
chart:
  appVersion: v1.2.36
tests:
  - it: telegramPolling.enabled=false -> should not create deployment (default)
    asserts:
      - hasDocuments:
          count: 0

  - it: telegramPolling.enabled=true -> should create telegram polling deployment
    set:
      telegramPolling.enabled: true
    asserts:
      - containsDocument:
          kind: Deployment
          apiVersion: apps/v1
          metadata.name: oncall-telegram-polling
      - isSubset:
          path: metadata.labels
          content:
            app.kubernetes.io/component: telegram-polling
            app.kubernetes.io/instance: oncall
            app.kubernetes.io/name: oncall
      - isSubset:
          path: spec.selector.matchLabels
          content:
            app.kubernetes.io/component: telegram-polling
            app.kubernetes.io/instance: oncall
            app.kubernetes.io/name: oncall
      - isSubset:
          path: spec.template.metadata.labels
          content:
            app.kubernetes.io/component: telegram-polling
            app.kubernetes.io/instance: oncall
            app.kubernetes.io/name: oncall
      # Should contain only one replica to avoid Conflict while polling Telegram updates
      - equal:
          path: spec.replicas
          value: 1
      - equal:
          path: spec.template.spec.serviceAccountName
          value: oncall
      - contains:
          path: spec.template.spec.initContainers
          content:
            name: wait-for-db
          any: true
      - matchSnapshot:
          path: spec.template.spec.containers

  - it: telegramPolling.resources -> should specify resources
    set:
      telegramPolling:
        enabled: true
        resources:
          limits:
            cpu: 100m
            memory: 128Mi
          requests:
            cpu: 100m
            memory: 128Mi
    asserts:
      - equal:
          path: spec.template.spec.containers[0].resources
          value:
            limits:
              cpu: 100m
              memory: 128Mi
            requests:
              cpu: 100m
              memory: 128Mi
