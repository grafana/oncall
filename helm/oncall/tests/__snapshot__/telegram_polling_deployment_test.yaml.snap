telegramPolling.enabled=true -> should create telegram polling deployment:
  1: |
    - command:
        - sh
        - -c
        - python manage.py start_telegram_polling
      env:
        - name: BASE_URL
          value: https://example.com
        - name: SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: SECRET_KEY
              name: oncall
        - name: MIRAGE_SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: MIRAGE_SECRET_KEY
              name: oncall
        - name: MIRAGE_CIPHER_IV
          value: 1234567890abcdef
        - name: DJANGO_SETTINGS_MODULE
          value: settings.helm
        - name: AMIXR_DJANGO_ADMIN_PATH
          value: admin
        - name: OSS
          value: "True"
        - name: DETACHED_INTEGRATIONS_SERVER
          value: "False"
        - name: UWSGI_LISTEN
          value: "1024"
        - name: BROKER_TYPE
          value: rabbitmq
        - name: GRAFANA_API_URL
          value: http://oncall-grafana
        - name: FEATURE_TELEGRAM_INTEGRATION_ENABLED
          value: "True"
        - name: FEATURE_TELEGRAM_LONG_POLLING_ENABLED
          value: "True"
        - name: TELEGRAM_WEBHOOK_HOST
          value: https://example.com
        - name: TELEGRAM_TOKEN
          value: ""
        - name: MYSQL_HOST
          value: oncall-mariadb
        - name: MYSQL_PORT
          value: "3306"
        - name: MYSQL_DB_NAME
          value: oncall
        - name: MYSQL_USER
          value: root
        - name: MYSQL_PASSWORD
          valueFrom:
            secretKeyRef:
              key: mariadb-root-password
              name: oncall-mariadb
        - name: REDIS_PROTOCOL
          value: redis
        - name: REDIS_HOST
          value: oncall-redis-master
        - name: REDIS_PORT
          value: "6379"
        - name: REDIS_DATABASE
          value: "0"
        - name: REDIS_USERNAME
          value: ""
        - name: REDIS_PASSWORD
          valueFrom:
            secretKeyRef:
              key: redis-password
              name: oncall-redis
        - name: RABBITMQ_USERNAME
          value: user
        - name: RABBITMQ_PASSWORD
          valueFrom:
            secretKeyRef:
              key: rabbitmq-password
              name: oncall-rabbitmq
        - name: RABBITMQ_HOST
          value: oncall-rabbitmq
        - name: RABBITMQ_PORT
          value: "5672"
        - name: RABBITMQ_PROTOCOL
          value: amqp
        - name: RABBITMQ_VHOST
          value: ""
      image: grafana/oncall:v1.2.36
      imagePullPolicy: Always
      name: telegram-polling
      securityContext: {}
